File: WinForms/CS/MVVMMainDemo/Modules/TodoApp/ViewModels/AppViewModel.cs
namespace TodoApp.ViewModels {
    using System.Threading.Tasks;
    using DevExpress.Mvvm;
    using DevExpress.Mvvm.POCO;
    using TodoApp.Data;

    public class AppViewModel {
        static AppViewModel() {
            // registering global service
            ServiceContainer.Default.RegisterService(new InMemoryRepository());
        }
        public AppViewModel() {
            Title = "Getting started witn MVVM - Todo App";
        }
        public string Title {
            get;
            private set;
        }
        protected INavigationService NavigationService {
            get { return this.GetService<INavigationService>(); }
        }
        public async Task OnShown() {
            // Show Logo screen
            NavigationService.Navigate("AppLogo", null, this, false);
            await Task.Delay(2000);
            // Show Items screen
            NavigationService.Navigate("ItemsView", null, this, true);
        }
    }
}

File: WinForms/VB/MVVMMainDemo/Modules/TodoApp/ViewModels/AppViewModel.vb
Imports System.Threading.Tasks
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports TodoApp.Data

Namespace TodoApp.ViewModels
    Public Class AppViewModel
        Private _Title As String
        Shared Sub New()
            ' registering global service
            ServiceContainer.Default.RegisterService(New InMemoryRepository())
        End Sub
        Public Sub New()
            Title = "Getting started witn MVVM - Todo App"
        End Sub
        Public Property Title As String
            Get
                Return _Title
            End Get
            Private Set(ByVal value As String)
                _Title = value
            End Set
        End Property
        Protected ReadOnly Property NavigationService As INavigationService
            Get
                Return GetService(Of INavigationService)()
            End Get
        End Property
        Public Async Function OnShown() As Task
            ' Show Logo screen
            NavigationService.Navigate("AppLogo", Nothing, Me, False)
            Await Task.Delay(2000)
            ' Show Items screen
            NavigationService.Navigate("ItemsView", Nothing, Me, True)
        End Function
    End Class
End Namespace
